roles/eda/templates/eda-webhook.deployment.yaml.j2

```diff
@@ -0,0 +1,235 @@
+    app.kubernetes.io/component: '{{ deployment_type }}-webhook'
+  name: '{{ webhook_server_name }}'
+  namespace: '{{ ansible_operator_meta.namespace }}'
+spec:
+  replicas: {{ combined_webhook.replicas }}
+{% if combined_webhook.strategy is defined %}
+  strategy:
+    type: {{ combined_webhook.strategy.type }}
+{% if combined_webhook.strategy.type == "Recreate" %}
+    rollingUpdate: null
+{% elif combined_webhook.strategy.type == "RollingUpdate" %}
+    rollingUpdate:
+      maxSurge:  {{ combined_webhook.strategy.rollingUpdate.maxSurge | default("25%")}}
+      maxUnavailable: {{ combined_webhook.strategy.rollingUpdate.maxUnavailable | default("25%")}}
+{% endif %}
+{% endif %}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: '{{ ansible_operator_meta.name }}'
+      app.kubernetes.io/managed-by: '{{ deployment_type }}-operator'
+      app.kubernetes.io/component: '{{ deployment_type }}-webhook'
+  template:
+    metadata:
+      labels:
+        {{ lookup("template", "../common/templates/labels/common.yaml.j2")  | indent(width=8) | trim }}
+        {{ lookup("template", "../common/templates/labels/version.yaml.j2") | indent(width=8) | trim }}
+        app.kubernetes.io/component: '{{ deployment_type }}-webhook'
+      annotations:
+        kubectl.kubernetes.io/default-container: 'eda-webhook'
+{% for template in [
+    "eda-webhook.configmap",
+  ] %}
+        checksum-{{ template | replace('/', '-') }}: "{{ lookup('template', template + '.yaml.j2') | sha1 }}"
+{% endfor %}
+{% for secret in [
+    "secrets/db_fields_encryption",
+  ] %}
+        checksum-secret-{{ secret }}: "{{ lookup('ansible.builtin.vars', secret, default='')["resources"][0]["data"] | default('') | sha1 }}"
+{% endfor %}
+    spec:
+      serviceAccountName: '{{ ansible_operator_meta.name }}'
+{% if image_pull_secrets | length > 0 %}
+      imagePullSecrets:
+{% for secret in image_pull_secrets %}
+        - name: {{ secret }}
+{% endfor %}
+{% endif %}
+{% if combined_webhook.node_selector is defined %}
+      nodeSelector:
+        {{ combined_webhook.node_selector | to_nice_yaml | indent(width=8) }}
+{% endif %}
+{% if combined_webhook.tolerations is defined %}
+      tolerations:
+        {{ combined_webhook.tolerations | to_nice_yaml | indent(width=8) }}
+{% endif %}
+{% if combined_webhook.topology_spread_constraints is defined %}
+      topologySpreadConstraints:
+        {{ combined_webhook.topology_spread_constraints | indent(width=8) }}
+{% endif %}
+      initContainers:
+      - name: wait-for-migrations
+        image: {{ _image }}
+        imagePullPolicy: '{{ image_pull_policy }}'
+        command:
+          - /bin/bash
+          - -c
+          - aap-eda-manage wait_for_migrations -t 60
+        envFrom:
+          - configMapRef:
+              name: '{{ ansible_operator_meta.name }}-{{ deployment_type }}-env-properties'
+        env:
+        - name: EDA_DB_HOST
```

Comment:
@rooftopcellist we need EDA_ACTIVATION_DB_HOST here which should be the same ad EDA_DB_HOST

Reply:
@mkanoor is this needed in other deplooyments? (default-worker and activation-worker)?  Or just this webhook one?

Also, does the webhook database instance share the pg credentials? or do we need to generate a separate PostgreSQL connect k8s secret?

Reply:
@rooftopcellist it is required for all, but I see you already added it to the them.